snapcraft/internal/lifecycle/_runner.py

```diff
@@ -316,6 +331,28 @@ def _handle_dirty(self, part, step, dirty_report, cli_config):
         getattr(self, '_re{}'.format(step.name))(part, hint='({})'.format(
             dirty_report.get_summary()))
 
+    def _handle_outdated(self, part, step, outdated_report, cli_config):
+        dirty_action = cli_config.get_outdated_step_action()
+        if not step.clean_if_dirty:
+            if dirty_action == config.OutdatedStepAction.ERROR:
+                raise errors.StepOutdatedError(
+                    step=step, part=part.name, outdated_report=outdated_report)
+
+        update_function = getattr(part, 'update_{}'.format(step.name), None)
+        if update_function:
+            self._prepare_to_run(step=step, part=part)
+            notify_part_progress(
+                part, 'Updating {} step for'.format(step.name), '({})'.format(
+                    outdated_report.get_summary()))
+            update_function()
+
+            # We know we just ran this step, so rather than check, manually
+            # twiddle the cache
+            self._step_complete(part, step)
+        else:
+            getattr(self, '_re{}'.format(step.name))(part, '({})'.format(
+                outdated_report.get_summary()))
+
 
 def notify_part_progress(part, progress, hint='', debug=False):
```

Comment:
this is old, but what is the difference?

Reply:
Just keep it green :smile:

Reply:
I'm afraid I'm not clear on what you're asking me to do, here.